lib/record_store/provider/ns1/client.rb

```diff
@@ -0,0 +1,42 @@
+require 'ns1'
+
+module RecordStore
+  class Provider::NS1 < Provider
+    class Error < StandardError; end
+    class Client < ::NS1::Client
+      def initialize(api_key:)
+        super(api_key)
+      end
+
+      def zones
+        super
+      end
+
+      def zone(name)
+        super(name)
+      end
+
+      def record(zone:, fqdn:, type:)
+        result = super(zone, fqdn, type)
+        return nil if result.is_a? NS1::Response::Error
```

Comment:
Should we return `nil` without even checking the kind of error ?
Is the absence of record the only kind of error possible to be returned by their API when fetching a record ?

Reply:
It seems weird but if you check `error.rb` of NS1 library, you might understand why we ended up approaching this way. 
https://github.com/kitop/ns1-ruby/blob/master/lib/ns1/response/error.rb

Reply:
We ended up enhancing record with an option that allows it to return nil for non-existent records.